apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.IntegrationTests/JobsTest.cs

```diff
@@ -67,5 +69,210 @@ public void ListJobs_FullProjection()
+                destinationUri, destinationReference,originTable.Schema,
+                new CreateLoadJobOptions() { SkipLeadingRows = 1});
+            loadJob = loadJob.PollUntilCompleted().ThrowOnAnyError();
+
+            var destinationTable = bqClient.GetTable(destinationReference);
+
+            Assert.Equal(originTable.ListRows().Count(), destinationTable.ListRows().Count());
+        }
+
+        [Fact]
+        public void LoadJob_Labels()
+        {
+            var bqClient = BigQueryClient.Create(_fixture.ProjectId);
+            var originTable = bqClient.GetTable(_fixture.DatasetId, _fixture.HighScoreTableId);
+            var destinationBucket = _fixture.StorageBucketName;
+            var destinationObject = _fixture.GenerateStorageObjectName();
+            var destinationUri = $"gs://{destinationBucket}/{destinationObject}";
+            var destinationReference = bqClient.GetTableReference(_fixture.DatasetId, _fixture.CreateTableId());
+
+            // Just extracting the data into GCS
+            var extractJob = originTable.CreateExtractJob(destinationUri);
+            extractJob = extractJob.PollUntilCompleted().ThrowOnAnyError();
+
+            var loadJob = bqClient.CreateLoadJob(
+                destinationUri, destinationReference, originTable.Schema, 
+                new CreateLoadJobOptions() { SkipLeadingRows = 1, Labels = JobLabels });
+            VerifyJobLabels(loadJob?.Resource?.Configuration?.Labels);
+
+            loadJob = loadJob.PollUntilCompleted().ThrowOnAnyError();
+            VerifyJobLabels(loadJob?.Resource?.Configuration?.Labels);
+        }
+
+        [Fact]
+        public void JobLabels_InvalidCharactersKey()
+        {
+            var client = BigQueryClient.Create(_fixture.ProjectId);
+            var table = client.GetTable(_fixture.ProjectId, _fixture.DatasetId, _fixture.HighScoreTableId);
+            var options = new QueryOptions()
+            {
+                Labels = new Dictionary<string, string>()
+                {
+                    {"This is Invalid & throws", "this_is_valid-2" }
+                }
+            };
+
+            Assert.ThrowsAny<GoogleApiException>(() => client.CreateQueryJob($"SELECT * FROM {table}", null, options));
+        }
+
+        [Fact]
+        public void JobLabels_InvalidCharactersValue()
+        {
+            var client = BigQueryClient.Create(_fixture.ProjectId);
+            var table = client.GetTable(_fixture.ProjectId, _fixture.DatasetId, _fixture.HighScoreTableId);
+            var options = new QueryOptions()
+            {
+                Labels = new Dictionary<string, string>()
+                {
+                    {"this_is_valid-2", "This is Invalid & throws" }
+                }
+            };
+
+            Assert.ThrowsAny<GoogleApiException>(() => client.CreateQueryJob($"SELECT * FROM {table}", null, options));
+        }
+
+        internal static IDictionary<string, string> JobLabels =>
```

Comment:
Could you move this to the top of the file? (We tend to write fields/properties, constructors, then methods. Sometimes the properties are after the constructor, and we're not terribly consistent on that.)

Reply:
Done.
At some point I though of adding that to the fixture (is used in the load test and the query test also) but wasn't sure about adding the VerifyJobLabels to the fixture as well